File: scripts/parse-abi.py
import json

with open('/vercel/share/v0-project/lib/contracts/abi-json/treasury-full.json', 'r') as f:
    abi = json.load(f)

print('=== CommitRevealTreasury Functions ===\n')
for item in abi:
    if item.get('type') == 'function':
        name = item.get('name', '')
        inputs = ', '.join([f"{i['type']} {i['name']}" for i in item.get('inputs', [])])
        outputs = ', '.join([o['type'] for o in item.get('outputs', [])])
        mutability = item.get('stateMutability', '')
        print(f"{name}({inputs}) {mutability} -> {outputs}")

print('\n=== Events ===\n')
for item in abi:
    if item.get('type') == 'event':
        name = item.get('name', '')
        inputs = ', '.join([f"{'indexed ' if i.get('indexed') else ''}{i['type']} {i['name']}" for i in item.get('inputs', [])])
        print(f"{name}({inputs})")

File: scripts/parse-abi.js
const fs = require('fs');

const abi = JSON.parse(fs.readFileSync('/vercel/share/v0-project/lib/contracts/abi-json/CommitRevealSession.json', 'utf8'));

console.log('=== Session ABI Functions ===\n');

abi.forEach(item => {
  if (item.type === 'function') {
    const inputs = item.inputs.map(i => `${i.type} ${i.name}`).join(', ');
    const outputs = item.outputs ? item.outputs.map(o => o.type).join(', ') : '';
    const mutability = item.stateMutability || '';
    console.log(`${item.name}(${inputs}) ${mutability} ${outputs ? `-> ${outputs}` : ''}`);
  }
});

console.log('\n=== Events ===\n');

abi.forEach(item => {
  if (item.type === 'event') {
    const inputs = item.inputs.map(i => `${i.indexed ? 'indexed ' : ''}${i.type} ${i.name}`).join(', ');
    console.log(`${item.name}(${inputs})`);
  }
});
